Fix javadoc warnings in telemetry_core

We have a lot of warnings emitted when running `./gradlew javadoc` in the `telemetry_core` module:

```
[snip]
/Users/jplumb/code/newrelic-telemetry-sdk-java/telemetry_core/src/main/java/com/newrelic/telemetry/SpanBatchSenderFactory.java:27: warning: no @return
  default SpanBatchSender createBatchSender(String apiKey) {
                          ^
/Users/jplumb/code/newrelic-telemetry-sdk-java/telemetry_core/src/main/java/com/newrelic/telemetry/SpanBatchSenderFactory.java:54: warning: no @param for apiKey
  default SenderConfigurationBuilder configureWith(String apiKey) {
                                     ^
/Users/jplumb/code/newrelic-telemetry-sdk-java/telemetry_core/src/main/java/com/newrelic/telemetry/SpanBatchSenderFactory.java:54: warning: no @return
  default SenderConfigurationBuilder configureWith(String apiKey) {
                                     ^
78 warnings
```

This could certainly be tackled with multiple PRs.

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/Attributes.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/Attributes.java
@@ -21,16 +21,29 @@ public class Attributes {
   /** Creates an empty object */
   public Attributes() {}
 
-  /** Creates a copy. Changes to the new copy will not affect the original and vice versa. */
+  /**
+   * @param original Creates a copy. Changes to the new copy will not affect the original and vice
+   *     versa.
+   */
   public Attributes(Attributes original) {
     rawAttributes.putAll(original.rawAttributes);
   }
 
-  /** Creates a copy. Changes to the new copy will not affect the original and vice versa. */
+  /**
+   * Creates a copy. Changes to the new copy will not affect the original and vice versa.
+   *
+   * @return new instance holding current attributes
+   */
   public Attributes copy() {
     return new Attributes(this);
   }
 
+  /**
+   * Merge attributes into current attributes
+   *
+   * @param incoming attributes to be merged
+   * @return this
+   */
   public Attributes putAll(Attributes incoming) {
     rawAttributes.putAll(incoming.rawAttributes);
     return this;
@@ -39,6 +52,8 @@ public class Attributes {
   /**
    * Add a string-valued attribute.
    *
+   * @param key to locate the value
+   * @param value to be stored
    * @return this
    */
   public Attributes put(String key, String value) {
@@ -49,6 +64,8 @@ public class Attributes {
   /**
    * Add a Number-valued attribute.
    *
+   * @param key to locate the value
+   * @param value to be stored
    * @return this
    */
   public Attributes put(String key, Number value) {
@@ -59,6 +76,8 @@ public class Attributes {
   /**
    * Add a boolean-valued attribute.
    *
+   * @param key to locate the value
+   * @param value to be stored
    * @return this
    */
   public Attributes put(String key, boolean value) {

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/EventBatchSenderFactory.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/EventBatchSenderFactory.java
@@ -23,6 +23,8 @@ public interface EventBatchSenderFactory {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return a new event batch sender instance
    */
   default EventBatchSender createBatchSender(String apiKey) {
     SenderConfigurationBuilder configuration =
@@ -50,6 +52,8 @@ public interface EventBatchSenderFactory {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return a new sender configuration instance
    */
   default SenderConfigurationBuilder configureWith(String apiKey) {
     return EventBatchSender.configurationBuilder().apiKey(apiKey).httpPoster(getPoster());

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/LogBatchSenderFactory.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/LogBatchSenderFactory.java
@@ -23,6 +23,8 @@ public interface LogBatchSenderFactory {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return a new log batch sender instance
    */
   default LogBatchSender createBatchSender(String apiKey) {
     SenderConfigurationBuilder configuration =
@@ -51,6 +53,8 @@ public interface LogBatchSenderFactory {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return a new sender configuration builder instance
    */
   default SenderConfigurationBuilder configureWith(String apiKey) {
     return LogBatchSender.configurationBuilder().apiKey(apiKey).httpPoster(getPoster());

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/MetricBatchSenderFactory.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/MetricBatchSenderFactory.java
@@ -23,6 +23,8 @@ public interface MetricBatchSenderFactory {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return a new metric batch sender instance
    */
   default MetricBatchSender createBatchSender(String apiKey) {
     SenderConfigurationBuilder configuration =
@@ -50,6 +52,8 @@ public interface MetricBatchSenderFactory {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return a new sender configuration builder instance
    */
   default SenderConfigurationBuilder configureWith(String apiKey) {
     return MetricBatchSender.configurationBuilder().apiKey(apiKey).httpPoster(getPoster());

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/SenderConfiguration.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/SenderConfiguration.java
@@ -71,6 +71,7 @@ public class SenderConfiguration {
     /**
      * Configure the New Relic Insert API key to use.
      *
+     * @param apiKey new relic api key
      * @return this builder;
      */
     public SenderConfigurationBuilder apiKey(String apiKey) {
@@ -78,6 +79,10 @@ public class SenderConfiguration {
       return this;
     }
 
+    /**
+     * @param httpPoster client responsible to execute the http request
+     * @return this builder
+     */
     public SenderConfigurationBuilder httpPoster(HttpPoster httpPoster) {
       this.httpPoster = httpPoster;
       return this;
@@ -132,6 +137,7 @@ public class SenderConfiguration {
      * Configure whether audit logging is enabled. Note: audit logging will log all data payloads
      * sent to New Relic at DEBUG level, in plain text.
      *
+     * @param auditLoggingEnabled if true, audit log will be enabled
      * @return this builder.
      */
     public SenderConfigurationBuilder auditLoggingEnabled(boolean auditLoggingEnabled) {
@@ -144,6 +150,7 @@ public class SenderConfiguration {
      * default User-Agent that the SDK sends, and is useful for monitoring various sources of data
      * coming into the New Relic systems.
      *
+     * @param secondaryUserAgent configure a secondary User-Agent
      * @return this builder.
      */
     public SenderConfigurationBuilder secondaryUserAgent(String secondaryUserAgent) {

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/SpanBatchSenderFactory.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/SpanBatchSenderFactory.java
@@ -23,6 +23,8 @@ public interface SpanBatchSenderFactory {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return a new instance of a span batch sender
    */
   default SpanBatchSender createBatchSender(String apiKey) {
     SenderConfigurationBuilder configuration =
@@ -50,6 +52,8 @@ public interface SpanBatchSenderFactory {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return a new instance of a span configuration builder
    */
   default SenderConfigurationBuilder configureWith(String apiKey) {
     return SpanBatchSender.configurationBuilder().apiKey(apiKey).httpPoster(getPoster());

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/events/Event.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/events/Event.java
@@ -22,6 +22,7 @@ public final class Event implements Telemetry {
    *
    * @param eventType The type of event.
    * @param attributes The key-value pairs that make up the event.
+   * @param timestamp time stamp in epoch milliseconds
    */
   public Event(String eventType, Attributes attributes, long timestamp) {
     if (eventType == null || "".equals(eventType)) {

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/http/HttpPoster.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/http/HttpPoster.java
@@ -14,7 +14,16 @@ import java.util.Map;
  */
 public interface HttpPoster {
 
-  /** Post data to the provided URL. */
+  /**
+   * Post data to the provided URL.
+   *
+   * @param url http url to be reached
+   * @param headers headers to be sent
+   * @param body body to be sent
+   * @param mediaType media type definition
+   * @return http response from the POST request
+   * @throws IOException in case of http request error
+   */
   HttpResponse post(URL url, Map<String, String> headers, byte[] body, String mediaType)
       throws IOException;
 }

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/logs/Log.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/logs/Log.java
@@ -27,32 +27,32 @@ public class Log implements Telemetry {
     this.throwable = throwable;
   }
 
-  /** The point in time (ms since UNIX epoch) that the log entry was created. */
+  /** @return The point in time (ms since UNIX epoch) that the log entry was created. */
   public long getTimestamp() {
     return timestamp;
   }
 
-  /** The log line itself. */
+  /** @return The log line itself */
   public String getMessage() {
     return message;
   }
 
-  /** Additional attributes associated with the log entry. */
+  /** @return Additional attributes associated with the log entry. */
   public Attributes getAttributes() {
     return attributes;
   }
 
-  /** The name of the service which produced this log entry. */
+  /** @return The name of the service which produced this log entry. */
   public String getServiceName() {
     return serviceName;
   }
 
-  /** The log level (eg. INFO, DEBUG, etc) for the log entry. */
+  /** @return The log level (eg. INFO, DEBUG, etc) for the log entry. */
   public String getLevel() {
     return level;
   }
 
-  /** Create a builder for building a new log entry. */
+  /** @return Create a builder for building a new log entry. */
   public static LogBuilder builder() {
     return new LogBuilder();
   }
@@ -134,42 +134,60 @@ public class Log implements Telemetry {
     private String level;
     private Throwable throwable;
 
-    /** The point in time (ms since UNIX epoch) that the log entry was created. */
+    /**
+     * @param timestamp The point in time (ms since UNIX epoch) that the log entry was created.
+     * @return log builder instance
+     */
     public LogBuilder timestamp(long timestamp) {
       this.timestamp = timestamp;
       return this;
     }
 
-    /** The log line itself. */
+    /**
+     * @param message The log line itself.
+     * @return log builder instance
+     */
     public LogBuilder message(String message) {
       this.message = message;
       return this;
     }
 
-    /** Additional attributes associated with the log entry. */
+    /**
+     * @param attributes Additional attributes associated with the log entry.
+     * @return log builder instance
+     */
     public LogBuilder attributes(Attributes attributes) {
       this.attributes = attributes;
       return this;
     }
 
-    /** The name of the service which produced this log entry. */
+    /**
+     * @param serviceName The name of the service which produced this log entry.
+     * @return log builder instance
+     */
     public LogBuilder serviceName(String serviceName) {
       this.serviceName = serviceName;
       return this;
     }
 
-    /** The log level (eg. INFO, DEBUG, etc) for the log entry. */
+    /**
+     * @param logLevel The log level (eg. INFO, DEBUG, etc) for the log entry.
+     * @return log builder instance
+     */
     public LogBuilder level(String logLevel) {
       this.level = logLevel;
       return this;
     }
 
-    /** Create the new {@link Log} entry. */
+    /** @return Create the new {@link Log} entry. */
     public Log build() {
       return new Log(this, throwable);
     }
 
-    /** Will assign a throwable to the log entry. */
+    /**
+     * @param e Will assign a throwable to the log entry.
+     * @return log builder instance
+     */
     public LogBuilder throwable(Throwable e) {
       Utils.verifyNonNull(e);
       this.throwable = e;

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/logs/LogBatchSender.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/logs/LogBatchSender.java
@@ -84,6 +84,7 @@ public class LogBatchSender {
   /**
    * Build the final {@link LogBatchSender}.
    *
+   * @param configuration new relict rest api ingest configurations
    * @return the fully configured LogBatchSender object
    */
   public static LogBatchSender create(SenderConfiguration configuration) {

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatch.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatch.java
@@ -35,6 +35,7 @@ public class MetricBatch extends TelemetryBatch<Metric> {
      * be included in all common attributes.
      *
      * @param serviceName - The name of the service
+     * @return this builder
      */
     public Builder serviceName(String serviceName) {
       commonAttributesBuilder.serviceName(serviceName);
@@ -47,6 +48,7 @@ public class MetricBatch extends TelemetryBatch<Metric> {
      * included in all common attributes as "instrumentation.provider".
      *
      * @param instrumentationProvider - The instrumentation provider name
+     * @return this builder
      */
     public Builder instrumentationProvider(String instrumentationProvider) {
       commonAttributesBuilder.instrumentationProvider(instrumentationProvider);

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatchSender.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatchSender.java
@@ -44,6 +44,7 @@ public class MetricBatchSender {
    *
    * <p>To be removed in 0.8.0
    *
+   * @return a new metric batch sender builder
    * @deprecated Use the {@link #create(SenderConfiguration)} method instead.
    */
   @Deprecated
@@ -93,6 +94,7 @@ public class MetricBatchSender {
   /**
    * Build the final {@link MetricBatchSender}.
    *
+   * @param configuration new relict rest api ingest configurations
    * @return the fully configured MetricBatchSender object
    */
   public static MetricBatchSender create(SenderConfiguration configuration) {

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatchSenderBuilder.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatchSenderBuilder.java
@@ -50,6 +50,8 @@ public class MetricBatchSenderBuilder {
    * Turns on audit logging. Payloads sent will be logged at the DEBUG level. Please note that if
    * your payloads contain sensitive information, that information will be logged wherever your logs
    * are configured.
+   *
+   * @return this builder
    */
   public MetricBatchSenderBuilder enableAuditLogging() {
     configBuilder.auditLoggingEnabled(true);
@@ -62,6 +64,8 @@ public class MetricBatchSenderBuilder {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return this builder
    */
   public MetricBatchSenderBuilder apiKey(String apiKey) {
     configBuilder.apiKey(apiKey);
@@ -71,6 +75,9 @@ public class MetricBatchSenderBuilder {
   /**
    * Provide an implementation for HTTP POST. {@link #build()} will throw if an implementation is
    * not provided or this method is not called.
+   *
+   * @param httpPoster client responsible to execute the http request
+   * @return this builder
    */
   public MetricBatchSenderBuilder httpPoster(HttpPoster httpPoster) {
     configBuilder.httpPoster(httpPoster);
@@ -80,6 +87,10 @@ public class MetricBatchSenderBuilder {
   /**
    * Provide additional user agent information. The product is required to be non-null and
    * non-empty. The version is optional, although highly recommended.
+   *
+   * @param product to be used in the secondary user agent
+   * @param version to be used in the secondary user agent
+   * @return this builder
    */
   public MetricBatchSenderBuilder secondaryUserAgent(String product, String version) {
     if (version == null || version.isEmpty()) {

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBuffer.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBuffer.java
@@ -141,6 +141,7 @@ public final class MetricBuffer {
      * be included in all common attributes as "service.name".
      *
      * @param serviceName - The name of the service
+     * @return this builder
      */
     public Builder serviceName(String serviceName) {
       commonAttributesBuilder.serviceName(serviceName);
@@ -154,6 +155,7 @@ public final class MetricBuffer {
      * provider will be included in all common attributes as "instrumentation.provider".
      *
      * @param instrumentationProvider - The name of the instrumentation library
+     * @return this builder
      */
     public Builder instrumentationProvider(String instrumentationProvider) {
       commonAttributesBuilder.instrumentationProvider(instrumentationProvider);

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/spans/SpanBatchSender.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/spans/SpanBatchSender.java
@@ -48,6 +48,7 @@ public class SpanBatchSender {
    *
    * <p>To be removed in 0.8.0
    *
+   * @return new instance of a span batch sender builder
    * @deprecated Use the {@link #create(SenderConfiguration)} method instead.
    */
   @Deprecated
@@ -96,6 +97,7 @@ public class SpanBatchSender {
   /**
    * Build the final {@link SpanBatchSender}.
    *
+   * @param configuration new relict rest api ingest configurations
    * @return the fully configured SpanBatchSender object
    */
   public static SpanBatchSender create(SenderConfiguration configuration) {

--- a/telemetry_core/src/main/java/com/newrelic/telemetry/spans/SpanBatchSenderBuilder.java
+++ b/telemetry_core/src/main/java/com/newrelic/telemetry/spans/SpanBatchSenderBuilder.java
@@ -51,6 +51,8 @@ public class SpanBatchSenderBuilder {
    * Turns on audit logging. Payloads sent will be logged at the DEBUG level. Please note that if
    * your payloads contain sensitive information, that information will be logged wherever your logs
    * are configured.
+   *
+   * @return this builder
    */
   public SpanBatchSenderBuilder enableAuditLogging() {
     configBuilder.auditLoggingEnabled(true);
@@ -63,6 +65,8 @@ public class SpanBatchSenderBuilder {
    * @see <a
    *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
    *     Relic API Keys</a>
+   * @param apiKey new relic api key
+   * @return this builder
    */
   public SpanBatchSenderBuilder apiKey(String apiKey) {
     configBuilder.apiKey(apiKey);
@@ -72,6 +76,9 @@ public class SpanBatchSenderBuilder {
   /**
    * Provide an implementation for HTTP POST. {@link #build()} will throw if an implementation is
    * not provided or this method is not called.
+   *
+   * @param httpPoster client responsible to execute the http request
+   * @return this builder
    */
   public SpanBatchSenderBuilder httpPoster(HttpPoster httpPoster) {
     configBuilder.httpPoster(httpPoster);
@@ -81,6 +88,10 @@ public class SpanBatchSenderBuilder {
   /**
    * Provide additional user agent information. The product is required to be non-null and
    * non-empty. The version is optional, although highly recommended.
+   *
+   * @param product to be used in the secondary user agent
+   * @param version to be used in the secondary user agent
+   * @return this builder
    */
   public SpanBatchSenderBuilder secondaryUserAgent(String product, String version) {
     Utils.verifyNonBlank(product, "Product cannot be null or empty in the secondary user-agent.");

